docs-en/chapter_computational_complexity/iteration_and_recursion.md

```diff
@@ -1,194 +1,193 @@
+For example, in the following code, the condition variable $i$ is updated twice in each iteration, a scenario that would be less convenient to implement with a `for` loop.
 
 ```src
 [file]{iteration}-[class]{}-[func]{while_loop_ii}
 ```
 
-Overall, **`for` loops have more compact code and `while` loops are more flexible**, and both can implement iteration structures. The choice of which one to use should be based on the needs of the particular problem.
+Overall, **`for` loops tend to be more concise, while `while` loops offer more flexibility**. Both can be used to implement iterative structures, and the choice between them should be based on the specific requirements of the problem at hand.
 
 ### Nested Loops
 
-We can nest one loop structure inside another, using the `for` loop as an example:
+Nested loops involve placing one loop structure within another. Taking `for` loops as an example:
 
 ```src
 [file]{iteration}-[class]{}-[func]{nested_for_loop}
 ```
 
-The figure below gives the block diagram of the flow of this nested loop.
+The figure below provides the flowchart for such a nested loop structure.
 
-![Block diagram of the flow of nested loops](iteration_and_recursion.assets/nested_iteration.png)
+![Flowchart of the Nested Loop](iteration_and_recursion.assets/nested_iteration.png)
 
-In this case, the number of operations of the function is proportional to $n^2$, or the algorithm's running time is "squared" to the size of the input data $n$.
+In such cases, the number of operations of the function is proportional to $n^2$, meaning the algorithm's runtime and the size of the input data $n$ have a 'quadratic relationship.'
 
-We can continue to add nested loops, and each nest is a "dimension up", which will increase the time complexity to "cubic relations", "quadratic relations", and so on.
+We can further increase the complexity by adding more nested loops, each level of nesting effectively "increasing the dimension," which raises the time complexity to "cubic," "quartic," and so on.
 
 ## Recursion
 
- "Recursion recursion is an algorithmic strategy to solve a problem by calling the function itself. It consists of two main phases.
+Recursion is an algorithmic strategy where a function solves a problem by calling itself. It primarily involves two phases:
 
-1. **recursive**: the program calls itself deeper and deeper, usually passing smaller or simpler arguments, until a "termination condition" is reached.
-2. **Recursion**: After the "termination condition" is triggered, the program returns from the deepest level of the recursion function, level by level, aggregating the results of each level.
+1. **Recursive Descent**: This is where the program repeatedly calls itself, often with progressively smaller or simpler arguments, moving towards the "base case" or "termination point." This phase is characterized by the function venturing deeper into successive levels of recursion.
```

Comment:
Remove "This phase is characterized by the function venturing deeper into successive levels of recursion." because it is not consistent with the Chinese docs.